### ISSUE TITLE
Using Equates for normal or inverted bitmasks in reassembly?

### ISSUE THREAD

**Issue by hwrobel** [NationalSecurityAgency/ghidra#3532]
I am looking at assembly code which uses a lot of bitmasks, e.g., 0x40 and 0x80 as individual status bits and in any combination in registers and immediates.

I looked for a method to have Ghidra show assembly instructions with immediate fields using bitmasks instead of simple values and couldn't find one, e.g.,

```
andinstruction register,#0xc0

==>

andinstruction register,#MyMask80|MyMask40
```

rather than having to define combination equates for all possible combinations, e.g., in this case MyMask80and40 (being an Equate for 0xc0) because there appears to be only a single Equate per value possible.
The immediate value would be augmented by the user specifying >=1 suitable Equates that would then as a displayed group form the value with any |0xHEX residue if needed, e.g., with appropriate Equate selection by the user:

```
andinstruction register,#0xd3

==>

andinstruction register,#MyMask80|MyMask40|0x13

```

The next step would be to have a selectable complete bit inversion for the value to permit selection from negated Equates for  the bit masks:

```
andinstruction register,#0x3f

==>

andinstruction register,#~(MyMask80|MyMask40)

```
In other words, I would like to use >=1 Equates to either define just all set bits in a value or just all cleared bits in a value.
Is anything like that possible?
Thanks.
